required/latex-lab/latex-lab-minipage.dtx

```diff
@@ -0,0 +1,202 @@
+%
+% \newcommand{\xt}[1]{\textsl{\textsf{#1}}}
+% \newcommand{\TODO}[1]{\textbf{[TODO:} #1\textbf{]}}
+% \newcommand{\docclass}{document class \marginpar{\raggedright document class
+% customizations}}
+%
+% \providecommand\hook[1]{\texttt{#1}}
+%
+% \begin{abstract}
+% \end{abstract}
+% 
+% \section{Introduction}
+% 
+%  How to tag minipages and parboxes is still under discussion. 
+%  But as they start and contain paragraphs it is clear that 
+%  hooks are needed to interrupt and restart the tagging. 
+%  
+%  The tagging added is currently quite simple minded. See the todo-testfiles and the
+%  code comments for some open points. 
+%  
+%    \begin{macrocode}
+%<*package>
+%    \end{macrocode}
+%
+% \section{Implementation}
+%    \begin{macrocode}
+\ProvidesExplPackage {latex-lab-testphase-minipage} {2023-05-10} {0.81}
+  {Code related to the tagging of minipages and parboxes}
+%    \end{macrocode}
+%  
+
+% \subsection{Provide the kernel hooks}
+% [kernel?]
+% We need to check/set the status before the
+% initial \cs{leavevmode} to decide how to handle the structure. 
+% Perhaps the same command will work for both?
+%    \begin{macrocode}
+\def\@kernel@init@minipage{}
+\def\@kernel@init@parbox{}
+%    \end{macrocode}
+% We need a few hooks.
+%    \begin{macrocode}
+\def\@kernel@before@minipage{}
+\def\@kernel@after@minipage{}
+\def\@kernel@before@parbox{}
+\def\@kernel@after@parbox{}
+%    \end{macrocode}
+%
+% \subsection{Patch minipage}
+% [kernel?]
+% The begin hook is inserted in \cs{@iiiminipage}
+%    \begin{macrocode}
+\def\@iiiminipage#1#2[#3]#4{%
+%    \end{macrocode}
+% Before switching to hmode we insert a hook. It is currently unused
+% but could for example check if we are in vmode and change the following
+% paragraph tag. 
+%    \begin{macrocode}
+  \@kernel@init@minipage
+  \leavevmode 
+  \@pboxswfalse
+  \setlength\@tempdima{#4}%
+  \def\@mpargs{{#1}{#2}[#3]{#4}}%
+%    \end{macrocode}
+% insert the hook
+%    \begin{macrocode}
+  \@kernel@before@minipage% new
+  \setbox\@tempboxa\vbox\bgroup
+    \color@begingroup
+      \hsize\@tempdima
+      \textwidth\hsize \columnwidth\hsize
+      \@parboxrestore
+      \def\@mpfn{mpfootnote}\def\thempfn{\thempfootnote}\c@mpfootnote\z@
+      \let\@footnotetext\@mpfootnotetext
+      \let\@listdepth\@mplistdepth \@mplistdepth\z@
+      \@minipagerestore
+      \@setminipage}
+%    \end{macrocode}
+%  And a hook at the end
+%    \begin{macrocode}
+\def\endminipage{%
+    \par
+    \unskip
+    \ifvoid\@mpfootins\else
+      \vskip\skip\@mpfootins
+      \normalcolor
+      \footnoterule
+      \unvbox\@mpfootins
+    \fi
+    \@minipagefalse   %% added 24 May 89
+  \color@endgroup
+  \egroup
+%    \end{macrocode}
+% as the parbox command is called, we must prevent that it adds its additional
+% tagging commands:
+%    \begin{macrocode}
+  \tag_stop:n{minipage}
+  \expandafter\@iiiparbox\@mpargs{\unvbox\@tempboxa}%
+  \tag_start:n{minipage}
+  \@kernel@after@minipage%new end ...
```

Comment:
I wonder if this should be handled differently. The kernel hooks could all be moved to the kernel code but  this `\tag_stop:n` business doesn't fit on this level. So perhaps another `\@kernel@..`would be more appropriate?

Reply:
This is naturally only a first draft for possible coding, but why do you think commands controlling tagging shouldn't be in the kernel (not directly naturally, but long term)?

Reply:
I think on that level it should all be \@kernel... without a mix using expl level coding. That's all.